### PR TITLE
[expo-location][ios] `startLocationUpdatesAsync` should not require background permissions

### PR DESCRIPTION
**Pull request by andrejpavlovic**
# Why

It looks like migration of code to Swift caused a regression where a background location permissions check was added to `startLocationUpdatesAsync` where it was not required before.

Before the rewrite (good):
https://github.com/expo/expo/blob/a2eab18fa91f04e884e2ea90079c3a28e85b5617/packages/expo-location/ios/EXLocation/EXLocation.m#L398-L405

After the rewrite (regression):
https://github.com/expo/expo/blob/b68e5216d26513415934ce28b2188442866a0167/packages/expo-location/ios/LocationModule.swift#L149-L152

# How

To reverse the regression, I've removed the background permission check and added back in the comments that were there in the pre-swift code.

# Test Plan

After requesting foreground permissions via `Location.useForegroundPermissions()`, calling `Location.startLocationUpdatesAsync` no longer throws `Background location permission is required to do this operation` error.

# Checklist

- [*] I added a `changelog.md` entry and rebuilt the package sources according to [this short guide](https://github.com/expo/expo/blob/main/CONTRIBUTING.md#-before-submitting)
- [*] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).
- [*] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)

### CODE REVIEW

File: packages/expo-location/CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,8 @@
 
 ### 🐛 Bug fixes
 
+- [iOS] `startLocationUpdatesAsync` should not require background permissions ([#33617](https://github.com/expo/expo/pull/33617) by [@andrejpavlovic](https://github.com/andrejpavlovic)
```
Comment (lukmccall) [inline packages/expo-location/CHANGELOG.md:11]:
```suggestion
- [iOS] `startLocationUpdatesAsync` should not require background permissions. ([#33617](https://github.com/expo/expo/pull/33617) by [@andrejpavlovic](https://github.com/andrejpavlovic)
```

**Review by lukmccall — APPROVED** [review]
Looks good! Thanks for your contribution.

**Comment by tyrauber** [comment]
Looks like this has been merged, but not released?  Would it be possible to cut 18.0.8 of expo-location? This is a blocker for fitness apps.  Thanks @andrejpavlovic for catching and fixing this.

**Comment by flodaniel** [comment]
Also still hoping this is released soon! I don't understand why a critical bug fix is unreleased for months.

**Comment by flodaniel** [comment]
@lukmccall is it possible to publish this as a bugfix release instead of working for the next expo SDK version?

**Comment by jasperaelvoet** [comment]
Still not released,

For now, you can install this specific commit using gitpkg:
`npm install https://gitpkg.vercel.app/expo/expo/packages/expo-location?3a92d4a`

**Pull request merged**